File: app/src/main/java/com/vpaliy/mediaplayer/di/module/ApplicationModule.java
package com.vpaliy.mediaplayer.di.module;


import android.content.Context;
import android.support.annotation.NonNull;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {

    private final Context context;

    public ApplicationModule(@NonNull Context context){
        this.context=context;
    }

    @Singleton
    @Provides
    public Context provideContext(){
        return context;
    }
}

File: app/src/main/java/com/vpaliy/mediaplayer/di/module/ApplicationModule.kt
package com.vpaliy.mediaplayer.di.module


import android.content.Context
import com.vpaliy.mediaplayer.domain.executor.BaseScheduler
import com.vpaliy.mediaplayer.domain.executor.SchedulerProvider
import com.vpaliy.mediaplayer.ui.base.Navigator
import javax.inject.Singleton
import dagger.Module
import dagger.Provides

@Module
class ApplicationModule(private val context: Context) {
    @Singleton
    @Provides
    internal fun provideContext()=context

    @Singleton
    @Provides
    internal fun provideScheduler():BaseScheduler=SchedulerProvider()

    @Singleton
    @Provides
    internal fun provideNavigator()=Navigator()
}
